### DISCUSSION TITLE
求指教，linux系统，dns配置问题

### DISCUSSION THREAD

**Discussion by buitt34** [apernet/hysteria#400] [Q&A]
服务端和客户端dns要怎么配置？服务端配置dns的文件在哪里？客户端我用的是passwall或者clash，该去哪里配置dns？求大神指教，万分感激!

**Comment by haruue** [comment]
找不到服务端配置文件通常意味着你是使用一键安装脚本安装的 hysteria ， 你应该附上你使用的一键安装脚本信息（最简单地， 把你执行的那条一键命令发出来）

**Reply by buitt34** [reply]
多谢大佬，我是这个脚本 bash <(curl -fsSL https://git.io/hysteria.sh)

**Reply by haruue** [reply]
你使用的一键脚本是 [Hi_hysteria](https://github.com/emptysuns/Hi_Hysteria)
这个一键脚本的服务端配置文件在 `/etc/hihy/conf/hihyServer.json`

**Reply by buitt34** [reply]
你好，我把这个文件的resolver改成了我的dns服务器地址，接下来我要怎么做呢？我的客户端是软路由里的passwall或者clash，我试着使用了一下，dns还是没有改变，解锁不了奈飞，如果我把/etc/resolv.conf里面的dns改成我可以解锁奈飞的dns，好像就可以，但是这个文件每隔一段时间他又会还原成原来的dns地址，不知道怎么办，搞了几天了

**Reply by buitt34** [reply]
我现在的困境是hy不能用dns，而我修改系统的dns又因为DHCP的问题导致每隔一段时间他就重新还原。。。

**Reply by haruue** [reply]
弄了半天原来你只是想解锁 Netflix 啊。。
我这里有个脚本， 你把里面的 `10.11.6.1` 换成你自己的解锁 Netflix DNS， 在 OpenWRT 路由器上跑一下， 应该就可以。。。
```
uci del dhcp.@dnsmasq[0].server
uci add_list dhcp.@dnsmasq[0].server='/fast.com/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/netflix.com/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/netflix.net/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/nflxext.com/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/nflximg.com/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/nflximg.net/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/nflxso.net/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/nflxvideo.net/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/netflixdnstest0.com/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/netflixdnstest1.com/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/netflixdnstest2.com/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/netflixdnstest3.com/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/netflixdnstest4.com/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/netflixdnstest5.com/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/netflixdnstest6.com/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/netflixdnstest7.com/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/netflixdnstest8.com/10.11.6.1'
uci add_list dhcp.@dnsmasq[0].server='/netflixdnstest9.com/10.11.6.1'
uci commit
```

**Reply by buitt34** [reply]
多谢啦，我就只想解锁奈飞而已，我刚刚学几天，软路由也是刚刚到手的，万分感谢，我马上试试看

**Reply by buitt34** [reply]
我的dns是购买那种15块一个月的，要绑定一个IP才能使用，教程里都是绑用来做服务器的那个IP，我按照你说的在软路由运行了这些代码，但是我看奈飞还是没有解锁，那么我是不是要填我的公网ip才行？

**Reply by haruue** [reply]
这个居然能卖 15 块钱一个月。。。

如果你绑定的是你的服务器公网 IP， 那你应该在服务器上配置对应的 DNS。

先在你的路由器上执行下面的命令清除刚刚的设置
```
uci del dhcp.@dnsmasq[0].server
uci commit
```

然后你可以直接编辑你服务器上的 `/etc/resolv.conf` ， 你可以先删除这个文件再重新创建， 然后填入下面的内容
```
nameserver 你购买的DNS服务器地址
```
保存应该就可以了。

**Reply by haruue** [reply]
不过这样的话还是要求你在看 Netflix 的时候通过 socks5 或者 http 直连 Hysteria 客户端的对应服务端口才行（比如通过浏览器插件）。。 Passwall 不一定可以。

**Reply by buitt34** [reply]
好的，我再研究研究，多谢大佬抽时间解答，万分感激
